scheduling/tasks.go

```diff
@@ -0,0 +1,124 @@
+package scheduling
+
+import (
+	"encoding/json"
+	"fmt"
+	"time"
+)
+
+type TaskId int32 // need to determine how we want to create IDs
+type RunId int32  // need to determine how we want to create IDs
+type TaskType string
+
+const (
+	ResourceCreation TaskType = "ResourceCreation"
+	HealthCheck      TaskType = "HealthCheck"
+	Monitoring       TaskType = "Monitoring"
+)
+
+type TargetType string
+
+const (
+	ProviderTarget    TargetType = "provider"
+	NameVariantTarget TargetType = "name_variant"
+)
+
+type Provider struct {
+	Name       string     `json:"name"`
+	TargetType TargetType `json:"target_type"`
+}
+
+type NameVariant struct {
+	Name       string     `json:"name"`
+	TargetType TargetType `json:"target_type"`
+}
+
+func (p Provider) Type() TargetType {
+	return p.TargetType
+}
+
+func (nv NameVariant) Type() TargetType {
+	return nv.TargetType
+}
+
+type TaskTarget interface {
+	Type() TargetType
+}
+
+type TaskMetadata struct {
+	ID     TaskId     `json:"id"`
+	Name   string     `json:"name"`
+	Type   TaskType   `json:"type"`
+	Target TaskTarget `json:"target"`
+	Date   time.Time  `json:"date"`
+}
+
+func (t *TaskMetadata) getID() TaskId {
+	return t.ID
+}
+
+func (t *TaskMetadata) getName() string {
+	return t.Name
+}
+
+func (t *TaskMetadata) getTarget() TaskTarget {
```

Comment:
just Target()

Reply:
Then it becomes a duplicate of the Target field in the TaskMetadata struct

Reply:
The struct members should actually probably be lowercase which would resolve this